python/ray/tune/function_runner.py

```diff
@@ -485,16 +469,16 @@ def wrap_function(train_func, warn=True):
 
     func_args = inspect.getfullargspec(train_func).args
     use_checkpoint = detect_checkpoint_function(train_func)
-    if len(func_args) > 1:  # more arguments than just the config
-        if "reporter" not in func_args and not use_checkpoint:
-            raise ValueError(
-                "Unknown argument found in the Trainable function. "
-                "Arguments other than the 'config' arg must be one "
-                "of ['reporter', 'checkpoint_dir']. Found: {}".format(
-                    func_args))
+    use_config_single = detect_config_single(train_func)
+    use_reporter = detect_reporter(train_func)
+
```

Comment:
Isn't `train_func(config, checkpoint_dir, reporter)` valid as well? Should there be a third case detecting when both checkpoint_dir and reporter are passed in?

Reply:
No, `reporter` shouldn't work with checkpoint_dir (intentionally). Actually, we should just remove the option to use a reporter.

```
In [1]: def tune_(config, reporter, checkpoint_dir):
   ...:     pass
   ...:

In [2]: from ray import tune

In [3]: tune.run(tune_)

ray.tune.error.TuneError: Trial raised an exception. Traceback:
ray::ImplicitFunc.train() (pid=5960, ip=192.168.1.115)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/tune/function_runner.py", line 225, in run
    self._entrypoint()
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/tune/function_runner.py", line 288, in entrypoint
    self._status_reporter.get_checkpoint())
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/tune/function_runner.py", line 515, in _trainable_func
    output = train_func(config, reporter)
```